design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+
+  # Provider-Scope [Optional] - Defines claim scope ie: openid email profile
+  - !variable provider-scope
+
+  # Status Webservice for authenticator
+  - !webservice
+    id: status
+    annotations:
+      description: Status service to check that the authenticator is configured correctly
+ 
+  - !group
+    id: operators
+    annotations:
+      description: Group of users who can check the status of the authenticator
+ 
+  - !permit
+     role: !group operators
+     privilege: [ read ]
+     resource: !webservice status
+
+  - !group
+    id: users
+    annotations:
+      description: Group of users who can authenticate using the authn-oidc/<service-id> authenticator
+
+  - !permit
+    role: !group users
+    privilege: [ read, authenticate ]
+    resource: !webservice
+
+#---
+# Need to grant user members to role
+- !grant
+  members:
+  - !user testuser1@mycompany.com
+  role: !group conjur/authn-oidc/<service id>/users
+```
+
+
+## Implementation
+An existing workaround for the CA cert issue mentioned previously involves a user manually updating the 
+OpenSSL truststore in the container to include any custom CA certs. We can leverage this idea in Conjur 
+to do the same thing on the fly. A simple wrapper method which creates a temporary symlink in the OpenSSL 
+truststore to a tempfile containing the certificate content should be sufficient. Cleanup involves 
+removing the symlink, and ensuring that the tempfile has been cleaned up after code execution. Since we 
+are (temporarily) adding the cert to the existing truststore, it should not interfere with concurrent 
+HTTP connections which may rely on the default OpenSSL certs still being available.
+
+An example of what this wrapper method may look like:
+```ruby
+def with_temporary_cert_store(cert_string)
+  Dir.mktmpdir do |temp_dir|
+    # Write the certificate to a temporary file
+    temp_file = File.join(temp_dir, 'ca.pem')
+    File.write(temp_file, cert_string)
+
+    # Compute the X.509 hash of the certificate subject and create a symlink to the temp file
+    cert = OpenSSL::X509::Certificate.new(cert_string)
+    symlink_name = File.join(OpenSSL::X509::DEFAULT_CERT_DIR, "#{cert.subject.hash.to_s(16)}.0")
+    File.symlink(temp_file, symlink_name)
+
+    yield
+  ensure
+    File.unlink(symlink_name) if File.symlink?(symlink_name)
+  end
+end
+```
+
+And its usage:
+```ruby
+::Conjur::CertUtils.with_temporary_cert_store(@ca_cert) do
+  @discovered_provider = @open_id_discovery_service.discover!(@provider_uri)
+end
+```
+
+As for the `HTTPS_PROXY` issue, it appears to be fixed in a more recent version of OpenIDConnect gem which 
+migrated from the `httpclient` library to `faraday` for handling outbound connections. Therefore we should 
+simply be able to bump this gem version (assuming no new issues arise) in order to address this issue.
+
+The last major concern was around better debug logging. This can also be enabled from the Conjur side, 
+via a code snippet like below:
```

Comment:
Are these environment variables and is this something we want to add to docs or kb?
Would OpenIDConnect.logger help resolve this issue - https://ca-il-jira.il.cyber-ark.com:8443/browse/CNJR-2105

Reply:
This is the implementation that would likely go in an initializer in Conjur.

For a user this would be enabled by setting `CONJUR_LOG_LEVEL` to `debug` per the docs https://docs.cyberark.com/AAM-DAP/13.0/en/Content/Operations/Services/logs.htm#Enabledebuglogmessages